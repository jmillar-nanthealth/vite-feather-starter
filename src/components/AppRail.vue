<template>
  <FeatherNavigationRail>
    <template v-slot:header>
      <div class="user">
        <FeatherIcon :icon="account" class="user-icon" />
        <div>
          <p class="sub">Signed in as</p>
          <p class="user-name">User name</p>
        </div>
      </div>
    </template>
    <template v-slot:main>
      <FeatherRailItem href="#" :icon="app" title="App One" />
      <FeatherRailItem href="#" :icon="app" title="App Two" />
      <FeatherRailItem href="#" :icon="app" title="App Three" />
    </template>

    <FeatherRailGroup label="Help Navigation">
      <FeatherRailItem href="#" :icon="app" title="Feedback" />
      <FeatherRailItem href="#" :icon="app" title="Sign out" />
    </FeatherRailGroup>
  </FeatherNavigationRail>
</template>
<script lang="ts">
import {
  FeatherNavigationRail,
  FeatherRailItem,
  FeatherRailGroup,
} from "@featherds/navigation-rail";

import App from "@featherds/icon/medical/Hospital";
import Account from "@featherds/icon/action/AccountCircle";
import { FeatherIcon } from "@featherds/icon";
import { defineComponent } from "vue";
export default defineComponent({
  computed: {
    app() {
      return App;
    },
    account() {
      return Account;
    },
  },
  components: {
    FeatherNavigationRail,
    FeatherRailItem,
    FeatherRailGroup,
    FeatherIcon,
  },
});
</script>
<style lang="scss" scoped>
@import "@featherds/styles/mixins/typography";
@import "@featherds/styles/themes/variables";
.user {
  display: flex;
  align-items: center;

  .user-icon {
    height: 2rem;
    width: 2rem;
    color: var($primary);
    margin-right: 1rem;
  }
  .user-name {
    @include subtitle1;
  }
  .sub {
    @include caption;
  }
}
</style>
