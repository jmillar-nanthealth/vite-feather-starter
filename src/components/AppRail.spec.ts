import { shallowMount } from "@vue/test-utils";
import AppRail from "./AppRail.vue";

describe("App Rail", () => {
  it("should render", () => {
    const wrapper = shallowMount(AppRail);
    expect(wrapper.element).toMatchSnapshot();
  });
});
