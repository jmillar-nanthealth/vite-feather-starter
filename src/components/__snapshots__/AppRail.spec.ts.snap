// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Rail should render 1`] = `
<feather-navigation-rail-stub
  labels="[object Object]"
  modelvalue="false"
/>
`;
